components/LogSummary.js

```diff
@@ -0,0 +1,121 @@
+import moment from 'moment'
+import { Component } from 'react'
+import { withAuth } from 'use-auth0-hooks'
+
+import { secureFetch } from '../util'
+
+class LogSummary extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      logs: null,
+      logsError: null
+    }
+    // TODO fix babel plugin so we can use class properties
+    // https://babeljs.io/docs/en/babel-plugin-proposal-class-properties
+    // https://nextjs.org/docs/advanced-features/customizing-babel-config
+    this.handleFetchLogs = this.handleFetchLogs.bind(this)
+  }
+
+  async handleFetchLogs (force = false) {
+    const { logs, logsError } = this.state
+    if (!force && (logs || logsError)) {
+      return
+    }
+
+    const { accessToken } = this.props.auth
+    if (!accessToken) {
+      return
+    }
+    const fetchedLogs = await secureFetch(`${process.env.API_BASE_URL}/api/secure/logs?startDate=blah`, accessToken)
+    if (fetchedLogs) {
+      this.setState({ logs: fetchedLogs })
+    }
+  }
+
+  async componentDidMount () {
+    await this.handleFetchLogs()
+  }
+
+  async componentDidUpdate () {
+    await this.handleFetchLogs()
+  }
+
+  render () {
+    const { auth } = this.props
+    const { logs, logsError } = this.state
+    if (!auth.isAuthenticated) return null
+    const hasLogs = logs && logs.length > 0
+    return (
+      <div>
+        <h2>Log Summary</h2>
+        <button onClick={this.handleFetchLogs}>Fetch logs</button>
+        <a
+          target='_blank'
+          rel='noopener noreferrer'
+          href='https://console.aws.amazon.com/apigateway/home?region=us-east-1#/usage-plans'
+        >
+          Open AWS console
```

Comment:
Because the link opens the API gateway home and not the logs, this button should be under a separate heading from Log Summary.

Reply:
Or maybe linking to Cloudwatch log groups would be better? https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:log-groups

Reply:
@binh-dam-ibigroup, this is intended to show **request** logs. Not logs from cloudwatch. The request logs are stored in API Gateway. Maybe the word "log" is confusing. But this is supposed to show how many requests have been used for each API key.